ceph/ceph_admin/helper.py

```diff
@@ -831,6 +832,57 @@ def validate_log_file_after_enable(cls):
     return True
 
 
+def validate_log_rotate(cls):
+    """
+    Verify logrotation of cephadm.log file
+    Args:
+        cls: cephadm instance object
+
+    Returns:
+        boolean
+    """
+    # step 1: check for the files in current log directory
+    out = cls.installer.exec_command(cmd="ls /var/log/ceph", sudo=True)
+    files = re.split("\n| ", out[0])
+    logFileExists = [f for f in files if f == "cephadm.log"]
+    if not logFileExists:
+        LOG.error("Log rotation failed: Logs not present in /var/log/ceph folder")
+        return False
+
+    # step 2: apply logrotate on cephadm.log
+    out = cls.installer.exec_command(
+        cmd="logrotate --force /etc/logrotate.d/cephadm", sudo=True
+    )
+    # if logrotate doesn't result in logs being zipped return False
+    out = cls.installer.exec_command(cmd="ls /var/log/ceph", sudo=True)
+    files = re.split("\n| ", out[0])
+    logFileExists = [f for f in files if f == "cephadm.log"]
+    if logFileExists:
+        LOG.error("Log rotation failed while enabling rotate")
+        return False
+
+    # step 3: perform any operation that would result in cephadm log be written
+    out, _ = cls.shell(args=["ceph", "orch", "ps", "--format", "json"])
+    daemons = json.loads(out)
+    daemon = [d for d in daemons if "node-exporter" in d["daemon_name"]][0]
+    out, _ = cls.shell(
+        args=["ceph", "orch", "daemon", "redeploy", daemon["daemon_name"]]
+    )
+
+    # step 4: check new cephadm.log file is created
+    out = cls.installer.exec_command(cmd="ls /var/log/ceph", sudo=True)
+    # if new logs are not created return False
+    files = re.split("\n| ", out[0])
+    logFileExists = [f for f in files if f == "cephadm.log"]
+    if not logFileExists:
+        LOG.error(
+            "Log rotation failed: Logs not present in /var/log/ceph folder after logrotate"
+        )
+        return False
```

Comment:
In this case, add custom exception and raise same exception, it will be easy for debugging.

Reply:
At the method calling end it is expecting a boolean value true or false for validation of log rotate functionality, so I would prefer logging error to log.error and returning false instead of raising exception. It would be cleaner I feel.

Reply:
You can use below code to raise exception, this will add same error message but as an exception which will be more clear.
```
class LogRotationError(Exception):
    pass
.
.
.
            raise LogRotationError("Log rotation failed: Logs not present in /var/log/ceph folder after logrotate")
```

Reply:
This is just one of the many validations we are already doing in this file. Now if i have to add exceptions to all of them it would change the scope of this PR. I would prefer taking this up as a different task. Please let me know your thoughts.

Reply:
Since the calling script expects a bool val, I feel the current approach is good. Also in case if a custom exception has to be added, it would be better to have it in the test file (not mandatory, just a suggestion).

LGTM

Reply:
Thanks Pranav. @vamahaja If you think this is okay. could you please approve the changes?

Reply:
@pranavprakash20 @manasagowri Yes, script expects boot value but in this case it will simply log error and continue to next test case. 
But when this TC get called in between other TCs and get fails, it will be hard to check for error log. If we add an exception it will be highlighted in the logs and will get stacktrace as well which will be easy for debugging. 
Otherwise it works as expected.

Reply:
@vamahaja As discussed let us take this up as part of a different PR since the scope for this PR would be changed if we need to change all the validations.